uniauth/views.py

```diff
@@ -147,29 +159,46 @@ def login(request):
         if num_institutions == 1:
             return HttpResponseRedirect(institutions[0][2] + query_params)
 
+        elif api_mode:
+            # return institutions and queryparams to be displayed on custom UI login form
+            return JsonResponse({'institutions': institutions, 'queryparams': query_params})
         # Otherwise, render the page (without the Login form)
         else:
             return render(request, 'uniauth/login.html', context)
 
     # If we are displaying the login form, and it's
     # a POST request, attempt to validate the form
     elif request.method == "POST":
-        form = LoginForm(request, request.POST)
+        if request.META['CONTENT_TYPE'] == 'application/json':
+            # in API mode, form data will be sent through Json, which must be handled differently
+            form = LoginForm(request, json.loads(request.body.decode()))
```

Comment:
❓ I'm probably missing something obvious, but I'm confused why we wouldn't still POST to the login view, even using a JWT flow.